gk/main.c

```diff
@@ -236,10 +236,10 @@ extract_packet_info(struct rte_mbuf *pkt, struct ipacket *packet)
 
 		ip6_hdr = packet->l3_hdr;
 		packet->flow.proto = ETHER_TYPE_IPv6;
-		rte_memcpy(packet->flow.f.v6.src, ip6_hdr->src_addr,
-			sizeof(packet->flow.f.v6.src));
-		rte_memcpy(packet->flow.f.v6.dst, ip6_hdr->dst_addr,
-			sizeof(packet->flow.f.v6.dst));
+		rte_memcpy(packet->flow.f.v6.src.s6_addr, ip6_hdr->src_addr,
+			sizeof(packet->flow.f.v6.src.s6_addr));
+		rte_memcpy(packet->flow.f.v6.dst.s6_addr, ip6_hdr->dst_addr,
```

Comment:
Use type safe assignments here as well.

Reply:
The field ``src_addr`` in ipv6 header is defined as ``uint8_t src_addr [16]`` instead of the type ``struct in6_addr`` , so we may not apply the type assignment here.